index.js

```diff
@@ -161,6 +169,15 @@ ClusterManager.prototype._startMaster = function() {
 };
 
 /**
+ * Runs before exit callback and exits the master process.
+ * @param {Error} [err] Error that caused the master process to exit.
+ */
+ClusterManager.prototype._exitMaster = function (err) {
+  this.options.beforeExit(err);
```

Comment:
should this take a cb? otherwise we might exist if the `beforeExit` function is  async